File: java/easy/190_reverse_bits.java
package java.easy;

class ReverseBits {
    public int reverseBits(int n) {
        int result = 0;

        for (int i = 0; i < 32; i++) {
            result <<= 1;
            result = result | (n & 1);
            n >>= 1;
        }

        return result;
    }
}

File: go/easy/190_reverse_bits.go
// https://leetcode.com/problems/reverse-bits/
package easy

func reverseBits(num uint32) uint32 {
	result := uint32(0)

	for i := 0; i < 32; i++ {
		bit := (num >> 1) & 1
		result = result | bit<<(31-i)
	}

	return result
}
